frontend/i_types.py

```diff
@@ -0,0 +1,298 @@
+from abc import ABCMeta, abstractmethod
+
+class Type(metaclass=ABCMeta):
+    """This class is the top-parent of all possible inferred types of a Python program.
+    Every type has its own class that inherits this class.
+    """
+
+    @abstractmethod
+    def is_subtype(self, t):
+        """Check if this type is a subtype of the parameter type.
+
+        Arguments:
+            type (Type): The type to check the subtype relationship against.
+        """
+        pass
+
+
+    @abstractmethod
+    def get_name(self):
+        """Get the name of the type"""
+        pass
+
+    # Two types are considered identical if their names exactly match.
+    def __eq__(self, other):
+        return self.get_name() == other.get_name()
+
+    def __hash__(self):
+        return hash(self.get_name())
+
+class TNone(Type):
+
+    def is_subtype(self, t):
+        return True
+
+    def get_name(self):
+        return "None"
+
+# TODO: Numerics should be subtype of class
+class TBool(Type):
+
+    def is_subtype(self, t):
+        return isinstance(t, (TBool, TInt, TFloat))
```

Comment:
Yes I'm planning to do this when I implement the classes inference and subtype relationships, where the `object` will be represented by a class, and has a supertype relationship with all other types.